qa/tasks/cephfs/test_subvolume.py

```diff
@@ -0,0 +1,299 @@
+import logging
+
+from tasks.cephfs.cephfs_test_case import CephFSTestCase
+from teuthology.exceptions import CommandFailedError
+
+log = logging.getLogger(__name__)
+
+
+class TestSubvolume(CephFSTestCase):
+    CLIENTS_REQUIRED = 1
+    MDSS_REQUIRED = 1
+
+    def setUp(self):
+        super().setUp()
+
+    def tearDown(self):
+        super().tearDown()
+
+    def test_subvolume_move_out_file(self):
+        """
+        To verify that file can't be moved out of subvolume
+        """
+        self.mount_a.run_shell(['mkdir', 'group'])
+        self.mount_a.run_shell(['mkdir', 'group/subvol1'])
+        self.mount_a.run_shell(['setfattr', '-n', 'ceph.dir.subvolume',
+                                '-v', '1', 'group/subvol1'])
+        self.mount_a.run_shell(['mv', 'group/subvol1', 'group/subvol2'])
+
+        self.mount_a.run_shell(['touch', 'group/subvol2/file1'])
+
+        command_failed = False
+        # file can't be moved out of a subvolume
+        try:
+            self.mount_a.run_shell(['rename', 'group/subvol2/file1',
+                                    'group/file1', 'group/subvol2/file1'])
+        except CommandFailedError:
+            # expected
+            command_failed = True
+            pass
+
+        self.assertTrue(command_failed == True)
+
+        # clean up
+        self.mount_a.run_shell(['rm', '-rf', 'group'])
```

Comment:
You can move this into tearDown() as it's common among all the test methods?

Reply:
but there's no per-test `setUp`/`tearDown` methods available
also per test setUp and tearDown will get too cumbersome

Reply:
moved common code for test setup and cleanup in appropriate methods